src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt

```diff
@@ -24,749 +24,982 @@ import it.pagopa.ecommerce.eventdispatcher.repositories.TransactionsEventStoreRe
+                        this.validationServiceId = npgTransactionGatewayAuthorizationData.validationServiceId
+                        this.timestampOperation =
+                            OffsetDateTime.parse(
+                                authCompleted.transactionAuthorizationCompletedData.timestampOperation,
+                                DateTimeFormatter.ISO_OFFSET_DATE_TIME
+                            )
+                                .atZoneSameInstant(ZoneId.of("Europe/Paris"))
+                                .truncatedTo(ChronoUnit.SECONDS)
+                                .format(DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+                                .toString()
+                        this.email = it
+                    }
+                }
+                .map { Optional.of(it) }
+                .defaultIfEmpty(Optional.empty())
+
+        return additionalPaymentInformations.map {
+            MyBankClosePaymentRequestV2Dto().apply {
+                paymentTokens =
+                    authCompleted.paymentNotices.map { paymentNotice -> paymentNotice.paymentToken.value }
+                outcome = MyBankClosePaymentRequestV2Dto.OutcomeEnum.valueOf(transactionOutcome.name)
+                this.transactionId = transactionId.value()
+
+                if (transactionOutcome == ClosePaymentOutcome.OK) {
+                    timestampOperation =
+                        OffsetDateTime.parse(
+                            authCompleted.transactionAuthorizationCompletedData.timestampOperation
+                        )
+                    this.totalAmount = totalAmountEuro
+                    this.fee = feeEuro
+                    this.timestampOperation =
+                        OffsetDateTime.parse(
+                            authCompleted.transactionAuthorizationCompletedData.timestampOperation,
+                            DateTimeFormatter.ISO_OFFSET_DATE_TIME
+                        )
+                    idPSP = authCompleted.transactionAuthorizationRequestData.pspId
+                    paymentMethod = authCompleted.transactionAuthorizationRequestData.paymentTypeCode
+                    idBrokerPSP = authCompleted.transactionAuthorizationRequestData.brokerName
+                    idChannel = authCompleted.transactionAuthorizationRequestData.pspChannelCode
+                }
+
+                this.additionalPaymentInformations = it.orElse(null)
+                transactionDetails = closePaymentTransactionDetails
             }
-          } else {
-            mono { UserDto().apply { type = UserDto.TypeEnum.REGISTERED } }
-          }
-        } else {
-          Mono.error(
-            RuntimeException(
-              "Invalid user id null for transaction with clientId: [${baseTransaction.clientId}]"))
         }
-      }
-      else -> Mono.error(RuntimeException("Unhandled client id: [${baseTransaction.clientId}]"))
     }
+
+    private fun getOutcomePaymentGateway(
```

Comment:
```suggestion
    // FIXME: This method is used only for cards, remove Redirect branches
    private fun getOutcomePaymentGateway(
```
This is used only for cards, I would remove the sections for Redirect either in this PR or in another with a FIXME